src/line_launcher/duration_slider_listener.rs

```diff
@@ -0,0 +1,35 @@
+use crate::{config, config::midi::MidiSlider, midi, Message};
+use std::sync::{mpsc::Receiver, Arc, Mutex};
+use std::thread;
+use wmidi::Channel;
+
+pub fn listen_for_duration_control_changes(
+    midi_messages_receiver: Receiver<Message>,
+    value: Arc<Mutex<f64>>,
+) {
+    thread::spawn(move || {
+        let slider = MidiSlider {
+            channel: Channel::from_index(15).unwrap(),
+            control_change: config::midi::u8_to_control_function(28).unwrap(),
+        };
+
+        for midi_message in midi_messages_receiver.iter() {
+            let midi_messages = vec![midi_message];
```

Comment:
The helpers in `pedals` are oriented around multiple MIDI messages (since the consuming pattern there is in chunks along with audio chunks), but here it seems to want to be just an individual message at a time so it probably makes sense to "unwind" these helpers to just accept a single MIDI message rather than doing this clunky conversion to a single-item list here

Reply:
Updated to just operate on a single MIDI message at a time